core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -366,7 +366,9 @@ public FormIndex deleteRepeat(FormIndex index) {
         // update multiplicities of other child nodes
         for (int i = 0; i < parentElement.getNumChildren(); i++) {
             TreeElement child = parentElement.getChildAt(i);
-            if (child.getMult() > childMult) {
+            // We also need to check that this element matches the deleted element (besides multiplicity)
+            // in the case where the deleted repeat's parent isn't a subgroup
+            if (child.doFieldsMatch(deleteElement) && child.getMult() > childMult) {
```

Comment:
Took me a bit to understand what `child.doFieldsMatch(deleteElement)` is filtering in/out. Might help to mention repeat entry 'siblings'. That is this clause selects siblings of the deleted repeat entry and ignores other child of the parent element.

Maybe just lifting this whole for-loop into a helper called something like `reduceMultiplicityOfSiblingRepeatEntries` (or something shorter 😄 )